client/extensions/woocommerce/state/sites/settings/general/handlers.js

```diff
@@ -0,0 +1,59 @@
+/**
+ * Internal dependencies
+ */
+import { areSettingsGeneralLoaded } from 'woocommerce/state/sites/settings/general/selectors';
+import { dispatchRequest } from 'state/data-layer/wpcom-http/utils';
+import { http } from 'state/data-layer/wpcom-http/actions';
+import {
+	WOOCOMMERCE_SETTINGS_GENERAL_REQUEST,
+	WOOCOMMERCE_SETTINGS_GENERAL_RECEIVE,
+} from 'woocommerce/state/action-types';
+
+export const handleSettingsGeneralSuccess = ( { dispatch }, action, next, { data } ) => {
+	const { siteId } = action;
+	dispatch( {
+		type: WOOCOMMERCE_SETTINGS_GENERAL_RECEIVE,
+		siteId,
+		data,
+	} );
+	return next( action );
```

Comment:
As far as I could tell, this is still necessary, otherwise the meta from the data layer doesn't get set.